src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -104,6 +133,52 @@ def handle_result_for_project(
             ex=LAST_UPDATE_REDIS_TTL,
         )
 
+    def handle_result_for_project_auto_onboarding_mode(
+        self, project_subscription: ProjectUptimeSubscription, result: CheckResult
+    ):
+        if result["status"] == CHECKSTATUS_FAILURE:
+            redis = _get_cluster()
+            key = build_onboarding_failure_key(project_subscription)
+            pipeline = redis.pipeline()
+            pipeline.incr(key)
+            pipeline.expire(key, ONBOARDING_FAILURE_REDIS_TTL)
+            failure_count = pipeline.execute()[0]
+            if failure_count >= ONBOARDING_FAILURE_THRESHOLD:
+                # If we've hit too many failures during the onboarding period we stop monitoring
+                delete_project_uptime_subscription(
+                    project_subscription.project,
+                    project_subscription.uptime_subscription,
+                    modes=[ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING],
+                )
+                # Mark the url as failed so that we don't attempt to auto-detect it for a while
+                set_failed_url(project_subscription.uptime_subscription.url)
+                redis.delete(key)
+                metrics.incr("uptime.result_processor.autodetection.failed_onboarding")
+        elif result["status"] == CHECKSTATUS_SUCCESS:
+            assert project_subscription.date_added is not None
+            if (
+                datetime.now(timezone.utc) - ONBOARDING_MONITOR_PERIOD
+                > project_subscription.date_added
+            ):
```

Comment:
feels like it could be nice to name this condition
```suggestion
            has_graduated_onboarding = datetime.now(timezone.utc) - ONBOARDING_MONITOR_PERIOD
                > project_subscription.date_added

            if has_graduated_onboarding:
```

Reply:
Not sure if there's a lot of benefit here if it's used once, I can include a comment if the code is unclear though

Reply:
IMO the benifit is that the name documents it, so it's quicker to understand the condition.

I can just read

`if has_graduated_onboarding` vs having to understand the actual condition. It also gives a hint as to the value of `datetime.now(timezone.utc) - ONBOARDING_MONITOR_PERIOD > project_subscription.date_added` so when I'm reading that I already have a context clue for what I'm looking at.

A comment is fine too though sure